bfg9000/file_types.py

```diff
@@ -43,6 +43,13 @@ def link(self):
 class StaticLibrary(Library):
     pass
 
+class WholeArchive(StaticLibrary):
+    def __init__(self, lib):
+        build_inputs.Node.__init__(self)
+        if lib.creator:
```

Comment:
You can drop the if statement here. Even when it's false, you still get the expected behavior (`WholeArchive.creator` is None).